osh/cmd_parse.py

```diff
@@ -1161,6 +1161,11 @@ def ParseBraceGroup(self):
         ate = self._Eat(Id.Lit_LBrace)
         left = word_.BraceToken(ate)
 
+        return self._ParseBraceGroupAfterLBrace(left)
+
+    def _ParseBraceGroupAfterLBrace(self, left):
```

Comment:
Another way to factor this is to use have ParseBraceGroup take an Optional[Token] left

And then if it's passed, it will skip the first 2 lines

What do you think?

( Aside: I noticed  we parse doc comments ### in any brace group, whereas they're only for function bodies  ... we might want to clean that up)